python/glide-shared/glide_shared/config.py

```diff
@@ -107,23 +107,91 @@ def __init__(
         self.jitter_percent = jitter_percent
 
 
+class ServiceType(Enum):
+    """
+    Represents the types of AWS services that can be used for IAM authentication.
+    """
+
+    ELASTICACHE = 0
+    """Amazon ElastiCache service."""
+    MEMORYDB = 1
+    """Amazon MemoryDB service."""
+
+
+class IamAuthConfig:
+    """
+    Configuration settings for IAM authentication.
+
+    Attributes:
+        cluster_name (str): The name of the ElastiCache/MemoryDB cluster.
+        service (ServiceType): The type of service being used (ElastiCache or MemoryDB).
+        region (str): The AWS region where the ElastiCache/MemoryDB cluster is located.
+        refresh_interval_seconds (Optional[int]): Optional refresh interval in seconds for renewing IAM authentication tokens.
+            If not provided, defaults to 300 seconds (5 min).
+    """
+
+    def __init__(
+        self,
+        cluster_name: str,
+        service: ServiceType,
+        region: str,
+        refresh_interval_seconds: Optional[int] = 300,
+    ):
+        self.cluster_name = cluster_name
+        self.service = service
+        self.region = region
+        self.refresh_interval_seconds = refresh_interval_seconds
+
+
```

Comment:
Do we need to allow `refresh_interval_seconds` to be `None`? Can't we just allow it to fall-back to the default value? I can't think of any reason we would need to allow the user to explicitly set it to `None` in the constructor, for example?

```suggestion
class IamAuthConfig:
    """
    Configuration settings for IAM authentication.

    Attributes:
        cluster_name (str): The name of the ElastiCache/MemoryDB cluster.
        service (ServiceType): The type of service being used (ElastiCache or MemoryDB).
        region (str): The AWS region where the ElastiCache/MemoryDB cluster is located.
        refresh_interval_seconds (int]: Refresh interval in seconds for renewing IAM authentication tokens.
            If not provided, defaults to 300 seconds (5 min).
    """

    def __init__(
        self,
        cluster_name: str,
        service: ServiceType,
        region: str,
        refresh_interval_seconds: int = 300,
    ):
        self.cluster_name = cluster_name
        self.service = service
        self.region = region
        self.refresh_interval_seconds = refresh_interval_seconds
```

Reply:
The design is have the optional in the api to the core and the core sets the defaults
https://github.com/valkey-io/valkey-glide/blob/7cc4ff698595bcc25b09348f50763bdf53050401/glide-core/src/iam/mod.rs#L21

Reply:
In that case, why provide the default value of `300` in Python at all? Why not just make the value completely optional? Right now, we are _saying_ that it is optional, but, by providing a default value, we are ensuring it is always populated. We seem to be stuck halfway?